.github/workflows/ci.yml

```diff
@@ -132,8 +132,8 @@ jobs:
       - name: Install Rust
         uses: actions-rust-lang/setup-rust-toolchain@v1
 
-      - name: Install protobuf-compiler
-        run: sudo apt-get install -y protobuf-compiler
+      - name: Install Protoc
```

Comment:
Hi,
Are these changes enough for these requirements?
"
protobuf-compiler needs to be installed for all build steps and platforms. Currently main is broken because protobuf compiler isn't installed on all platforms.

Expectations

Builds should pass on all platforms.
Use https://github.com/marketplace/actions/setup-protoc instead of manually installing
Reduce duplication as much as possible in ci.yml
"
Thanks
@coderabbitai

Reply:
there is only one platform in which protoc step used in ci right?. Ubuntu-Latest. @coderabbitai

Reply:
But the issue author mentions all platforms why? @coderabbitai

Reply:
But the issue author mentions all platforms why? https://github.com/coderabbitai Explain

Reply:
But the issue author mentions all platforms why? Explain @coderabbitai

Reply:
Why release job is not using protoc? @coderabbitai

Reply:
Current
We can only setup one request handler in a JS worker file. This proposal intends to break that restriction

Proposal

type schema @upstream(onRequest: "onRequest") {
  # ... 
}

type Query {
  foo: Foo @http(path: "/foo", onRequest: "onRequestFoo")
  bar: Bar @http(path: "/bar", onRequest: "onRequestBar")
  baz: Bar @http(path: "/baz")
}
function onRequest (request) {
  // 
}

function onRequestFoo (request) {
  // 
}

function onRequestBar (request) {
  // 
}
In the above implementation when:

We query foo, onRequestFoo is called with the request params.
We query bar, onRequestBar is called with the request params.
We query baz, onRequest (default handler) is called with the request params.

and also look at this:
I think the user can have a logic like this in js worker.
pseudo code :

// onRequest being a generic handler/interceptor.
function onRequest ({request}) {
     // some logic to get path
     ......
    // switch case for path
    switch(request_path) {
        case "/foo":
            return onRequestFoo(request);
            break;
        case "/bar":
            return onRequestBar(request);
            break;
        default:
            // do nothing for others like Baz, just return request back to caller.
            return {request} ;
    }
}

function onRequestFoo ({request}) {
  //
}

function onRequestBar ({request}) {
  // 
}
This is very easy to user. Making onrequest a generic handler/interceptor.
No changes required in rust code.

What is the advantage the user gets if supposed to configure it in .graphql format?

Thanks. @coderabbitai

Reply:
one example. @coderabbitai

Reply:
@coderabbitai,
ok. now look at this.
look at command variable in on_request function in "src/cli/javascript/request_filter.rs" file.
in command variable decleration expression,
i want to pass dynamic handler name instead of a default "onRequest" string.
But how to get request handler name defined by user in .graphql format here and pass it to command expression to tell the worker to execute that handler function.

Reply:
@coderabbitai
ok. now look at this.
look at command variable in on_request function in "src/cli/javascript/request_filter.rs" file.
in command variable decleration expression,
i want to pass dynamic handler name instead of a default "onRequest" string.
But how to get request handler name defined by user in .graphql format here and pass it to command expression to tell the worker to execute that handler function.

Reply:
@coderabbitai,
I think we already had that logic. src/config/reader.rs